### DISCUSSION TITLE
JS beer.min.js functionality

### DISCUSSION THREAD

**Discussion by bariccattion** [beercss/beercss#119] [Q&A] [answered]
First and foremost, great library, I've been using it extensively for the past couple of weeks and it really speeds up development, it served all my purposes (except missing a DatePicker, but that was already marked as an improvement)
However, I have a question. 
Do we have a list, or section on the documentation, on what functionalities does the .js "beer.min.js" add?
I ask this because I've been using it alongside next.13 with Server Side Components, and would like to know what type of functionalities I can still expect in Server Components without including "beer.min.js", since it depends on the window (So that I can properly include it and use Client Components when needed)
🇧🇷

**Comment by leonardorafael** [comment, marked as answer]
Hi @bariccattion "obrigado pelo feedback" 💛

**Do we have a list, or section on the documentation, on what functionalities does the .js "beer.min.js" add?**
Yes, some elements have a section called "Trigger". Depending of the trigger method you used, you can load the js file of beercss in browser.

- https://github.com/beercss/beercss/blob/main/docs/DROPDOWN.md#triggers
- https://github.com/beercss/beercss/blob/main/docs/INPUT.md#triggers
- https://github.com/beercss/beercss/blob/main/docs/MODAL.md#triggers
- https://github.com/beercss/beercss/blob/main/docs/OVERLAY.md#triggers
- https://github.com/beercss/beercss/blob/main/docs/PAGE.md#triggers
- https://github.com/beercss/beercss/blob/main/docs/PROGRESS.md#triggers
- https://github.com/beercss/beercss/blob/main/docs/SELECT.md#triggers
- https://github.com/beercss/beercss/blob/main/docs/TABS.md#triggers
- https://github.com/beercss/beercss/blob/main/docs/TEXTAREA.md#triggers
- https://github.com/beercss/beercss/blob/main/docs/TOAST.md#triggers

The js file of beercss is not required, but I always recommend to load it in browser because:
- the size is very tiny, has about 2kb <img src="https://img.badgesize.io/beercss/beercss/main/dist/cdn/beer.min.js?compression=brotli">
- it does the job for you automatically, using the `data-ui` attribute
- write less js code in your project

**Reply by bariccattion** [reply]
Thanks for the response! I was just wondering if aside from the Triggers, we had any other functionality inside it
When I started using the library, I was just using the "active" class approach to triggering those elements, and just handling the state of those components by myself. However, I've found out that in some cases, using only that did not work properly. In most cases, the overlay behind the modal didn't show up.
When I really need to know the state of the component being triggered, I'm just using data-ui alongside MutationObserver, so I don't have to actively handle the state of the component by myself. data-ui handles triggering the component, and I just watch to see if that component is active. Not the best approach but still is framework agnostic, and I get the benefit of having a very well isolated component that I don't have to trigger by myself
